src/Analyzers/CSharp/CodeFixes/MakeMemberRequired/CSharpMakeMemberRequiredCodeFixProvider.cs

```diff
@@ -0,0 +1,133 @@
+
+            if (!CanBeAccessed(containingTypeVisibility, minimalAccessibility))
+                return;
+
+            RegisterCodeFix(context, CSharpCodeFixesResources.Make_property_required, nameof(CSharpCodeFixesResources.Make_property_required));
+        }
+        else if (fieldOrPropertySymbol is IFieldSymbol fieldSymbol)
+        {
+            var containingTypeVisibility = fieldSymbol.ContainingType.GetResultantVisibility();
+            var accessibility = fieldSymbol.DeclaredAccessibility;
+
+            if (!CanBeAccessed(containingTypeVisibility, accessibility))
+                return;
+
+            RegisterCodeFix(context, CSharpCodeFixesResources.Make_field_required, nameof(CSharpCodeFixesResources.Make_field_required));
+        }
+
+        // The `required` modifier cannot be used if a member is not accessible from outside of type.
+        // For instance, having private required property in a public class leads to compiler error.
+        // This function checks whether the member can be accessed by checking containing type visibility (which is computed already taking into account whether the type is nested and what is its visibility based on that fact)
+        // against accessibility of member we are trying to make required
+        static bool CanBeAccessed(SymbolVisibility containingTypeVisibility, Accessibility accessibility) => containingTypeVisibility switch
+        {
+            SymbolVisibility.Public => accessibility is Accessibility.Public,
+            SymbolVisibility.Internal => accessibility is >= Accessibility.Internal,
+            SymbolVisibility.Private => accessibility is >= Accessibility.Internal,
+            _ => throw ExceptionUtilities.Unreachable(),
+        };
+    }
+
+    protected override Task FixAllAsync(Document document, ImmutableArray<Diagnostic> diagnostics, SyntaxEditor editor, CodeActionOptionsProvider fallbackOptions, CancellationToken cancellationToken)
+    {
+        var root = editor.OriginalRoot;
+        var generator = editor.Generator;
+        var visitedFieldDeclarations = new HashSet<FieldDeclarationSyntax>();
+
+        foreach (var diagnostic in diagnostics)
+        {
+            var memberDeclarator = root.FindNode(diagnostic.Location.SourceSpan);
+
+            // If we are fixing field, do not apply new declaration modifiers just to variable declarator, but to the whole field declaration.
+            // This is observable when there are several variables in single filed declaration:
+            // `public string _myField, _myField1;` -> `public required string _myField, _myField1;`
+            // Without this branch the result would be:
+            // ```
+            // public required string _myField;
+            // public required string _myField2;
+            // ```
+            if (memberDeclarator is VariableDeclaratorSyntax { Parent.Parent: FieldDeclarationSyntax fieldDeclaration })
```

Comment:
needs a check at around line 50 that we have an appropriate declarator that we can eventually fix up here.  we have absolutely no guarantees about which nodes the compiler places diagnostics no.  only when ide owns analyzer/fixer pair can we make assumptions on the fix side since we can tightly ensure that they stay in sync.

Reply:
I got the idea but I still don't understand why we need that double-check. Imagine at some point compiler team decides to report this diagnostic at some other node. If they do, the'll make a PR to roslyn. At this poin all tests for an IDE fixer will break, no matter if they all crash or just don't produce the fix. Why should we care about fixer being crash-resistent if the break will be caught by the CI when trying to change compiler side anyway then?

Reply:
>  If they do, the'll make a PR to roslyn.

They definitely may not.  They may make a change that reports this on some new node we're not testing.  And they may not check any analyzers (they are not required to, and they may not even know the analyzer exists).  This is not speculative.  This has happened.

> Why should we care about fixer being crash-resistent if the break will be caught by the CI when trying to change compiler side anyway then?

Because CI only catches if we have the new use case checked by a test that exercises this codepath.  There is no guarantee of htat *at all*.  And, indeed, it's the *uncommon* case that that happens.